src/movement.cpp

```diff
@@ -923,11 +923,15 @@ bool MoveEvent::executeStep(Creature* creature, Item* item, const Position& pos)
 uint32_t MoveEvent::fireEquip(Player* player, Item* item, slots_t slot, bool isCheck)
 {
 	if (scripted) {
-		if (equipFunction(this, player, item, slot, isCheck) == 1) {
+		if (!equipFunction) {
 			if (executeEquip(player, item, slot, isCheck)) {
 				return 1;
-			} else {
-				return 0;
+			}
+		} else {
+			if (equipFunction(this, player, item, slot, isCheck) == 1) {
+				if (executeEquip(player, item, slot, isCheck)) {
+					return 1;
+				}
```

Comment:
Would anyone be against condensing this into a single if statement?
```c++
if (!equipFunction || equipFunction(this, player, item, slot, isCheck) == 1) {
    if (executeEquip(player, item, slot, isCheck)) {
        return 1;
    }
}
```

Reply:
I don't mind tbh, most of the time here it's rather done by a nested if, for better readability I guess, so I've just adapted to the convention of how it's mostly done.

Reply:
It can be condensed even further.

Reply:
I guess you mean like
```cpp
if ((!equipFunction) || (equipFunction(this, player, item, slot, isCheck) == 1) && (executeEquip(player, item, slot, isCheck))) {
        return 1;
}
```

Reply:
I'm more for readability but if both of you want it to condensed then say it please and I'll do it.
Because this should get merged already, as it fixes a crash which occurs on movements right now.

Reply:
I'll merge this if you apply this change.

Also couldn't you just condense it to the likes of this?
```CPP
if ((!equipFunction || equipFunction(this, player, item, slot, isCheck) == 1) && executeEquip(player, item, slot, isCheck)) {
        return 1;
}
```

Reply:
actually no, because we would skip `executeEquip(player, item, slot, isCheck` if `!equipFunction` would be false.
I'll stick with what @jo3bingham suggested, if that's ok?

Reply:
Sure, that's fine